pkg/util/taints/taints.go

```diff
@@ -192,7 +192,6 @@ func deleteTaints(taintsToRemove []v1.Taint, newTaints *[]v1.Taint) ([]error, bo
 	allErrs := []error{}
 	var removed bool
 	for _, taintToRemove := range taintsToRemove {
-		removed = false
```

Comment:
This is very suspicious! will have to look deeper

Reply:
the way it is now the last call to `DeleteTaint` or `DeleteTaintsByKey` determines whether we return true or false. I think we should return true if all the taints were deleted (i.e. no errors were encountered).

How about we switch the implementation to:
```
// deleteTaints deletes the given taints from the node's taintlist.
func deleteTaints(taintsToRemove []v1.Taint, newTaints *[]v1.Taint) ([]error, bool) {
	allErrs := []error{}
	for _, taintToRemove := range taintsToRemove {
		removed := false
		if len(taintToRemove.Effect) > 0 {
			*newTaints, removed = DeleteTaint(*newTaints, &taintToRemove)
		} else {
			*newTaints, removed = DeleteTaintsByKey(*newTaints, taintToRemove.Key)
		}
		if !removed {
			allErrs = append(allErrs, fmt.Errorf("taint %q not found", taintToRemove.ToString()))
		}
	}
	return allErrs, len(allErrs) == 0
}
```